accumulation_edit.php

```diff
@@ -0,0 +1,312 @@
+<?php
+/**
+ *	Accumulation Reports edit page.
+ */
+
+namespace Nottingham\AdvancedReports;
+
+const ACC_SMART_VAR = '[accumulation-number]';
+
+
+
+// Check user can edit accumulation reports and verify the report exists and is an accumulation
+// report.
+// Redirect to main reports page if not.
+$reportID = $_GET['report_id'];
+$listReports = $module->getReportList();
+if ( ! $module->isReportEditable( 'accumulation' ) ||
+     ! isset( $listReports[$reportID] ) || $listReports[$reportID]['type'] != 'accumulation' )
+{
+	header( 'Location: ' . $module->getUrl( 'reports.php' ) );
+	exit;
+}
+$reportConfig = $listReports[$reportID];
+$reportData = $module->getReportData( $reportID );
+$canSaveIfApi = ( ! $module->getSystemSetting( 'admin-only-api' ) ||
+                  $module->getUser()->isSuperUser() );
+$canSaveIfPublic = ( ! $module->getSystemSetting( 'admin-only-public' ) ||
+                     $module->getUser()->isSuperUser() );
+
+
+
+// Handle form submissions.
+if ( ! empty( $_POST ) )
+{
+	// Validate data
+	$validationMsg = '';
+	if ( ! $canSaveIfApi && $_POST['report_as_api'] == 'Y' )
+	{
+		$validationMsg = 'Reports with API access can only be saved by an administrator.';
+	}
+	if ( ! $canSaveIfPublic && $_POST['report_as_public'] == 'Y' )
+	{
+		$validationMsg = 'Reports with Public access can only be saved by an administrator.';
+	}
+	// - Check that the accumulation range settings are valid.
+	if ( $validationMsg == '' )
+	{
+		if ( ! \LogicTester::isValid( $_POST['acc_start'] ) )
+		{
+			$validationMsg = 'Start of accumulation range is not valid.';
+		}
+		elseif ( ! \LogicTester::isValid( $_POST['acc_end'] ) )
+		{
+			$validationMsg = 'End of accumulation range is not valid.';
```

Comment:
Do we check somewhere that start range < end range?

Reply:
This is not enforced because the accumulation numbers can be either incrementing or decrementing.